src/scales/scale.category.js

```diff
@@ -49,10 +50,19 @@ module.exports = Scale.extend({
 
 	getLabelForIndex: function(index, datasetIndex) {
 		var me = this;
-		var data = me.chart.data;
+		var chart = me.chart;
+		var data = chart.data;
 		var isHorizontal = me.isHorizontal();
+		var controller = chart.getDatasetMeta(datasetIndex).controller;
 
-		if (data.yLabels && !isHorizontal) {
+		// For controllers supporting getValueScaleId, we can be sure if this is a value scale.
+		// For object data ({x,y} etc.) we can rely on getRightValue to return correct thing.
+		// For others, check if data is a label and use that instead (so this is an value index, but values are labels)
```

Comment:
"an value" should be "a value".  ("an" is typically only used when a word starts with a, e, i, o, u)